src/Model/Response/ReferenceData.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\CdtClient\Model\Response;
+
+/**
+ * Class ReferenceData.
+ *
+ * Represents the business reference data received from the CDT API.
+ */
+class ReferenceData
+{
+    /**
+     * The list of departments.
+     */
+    protected ReferenceItemCollection $departments;
+
+    /**
+     * The list of priorities.
+     */
+    protected ReferenceItemCollection $priorities;
+
+    /**
+     * The list of purposes.
+     */
+    protected ReferenceItemCollection $purposes;
+
+    /**
+     * The list of delivery modes.
+     */
+    protected ReferenceItemCollection $deliveryModes;
+
+    /**
+     * The list of confidentialities.
+     */
+    protected ReferenceItemCollection $confidentialities;
+
+    /**
+     * The list of languages.
+     *
+     * @var array<int, string>
+     */
+    protected array $languages;
+
+    /**
+     * The list of statuses.
+     */
+    protected ReferenceItemCollection $statuses;
+
+    /**
+     * The list of services.
+     */
+    protected ReferenceItemCollection $services;
+
+    /**
+     * The list of send options.
+     */
+    protected ReferenceItemCollection $sendOptions;
+
+    /**
+     * The contacts for the translation service.
+     */
+    protected ReferenceContactCollection $contacts;
+
+    public function getDepartments(): ReferenceItemCollection
+    {
+        return $this->departments;
+    }
+
+    /**
+     * @param ReferenceItemCollection|array<int, array<string, string>> $departments
+     */
+    public function setDepartments(ReferenceItemCollection|array $departments): self
+    {
+        $this->departments = is_array($departments) ? new ReferenceItemCollection($departments) : $departments;
```

Comment:
Unfortunately I couldn't get it to work with just `ReferenceItemCollection`. I have tried the following approaches:

- Annotating properties as `@var ReferenceItemCollection $departments` creates an empty collection.
- Annotating properties as `@var ReferenceItemCollection<ReferenceItem> $departments` has no effect.
- Creating a custom denormalizer for the `BaseCollection` seems like a nice idea, but it requires some deep research if we want to have a generic solution for all collections. I haven't been able to come up with a better solution than we have now.
- Creating mutator methods (like `addDepartments`) makes the properties recognizable as collections, but it does not solve any problem. We still have the same exceptions
- Adding callbacks to `AbstractObjectNormalizer` could solve the problem, but it requires providing a list of fields to convert to collections. This would require a lot of hardcoding.

The major drawback of all these methods, including the current one, is that we get an array of arrays as collection contents. So instead of initializing the collection with `CollectionItem` items, we have to manually deserialize it from the raw array.

I didn't want to spend more time researching this, as it might make sense to finish all the endpoints and tests, and then try to change it in a test-driven way when we have some time left in the project.

Please let me know if you have any ideas to solve it better.